tests/laser/evm_testsuite/evm_test.py

```diff
@@ -64,22 +64,23 @@ def test_vmtest(
     # Act
     laser_evm.time = datetime.now()
 
-    # TODO: move this line below and check for VmExceptions when gas has been implemented
-    if post_condition == {}:
-        return
-
     execute_message_call(
         laser_evm,
         callee_address=action["address"],
         caller_address=action["caller"],
         origin_address=action["origin"],
         code=action["code"][2:],
-        gas=action["gas"],
+        gas_limit=int(action["gas"], 16),
         data=binascii.a2b_hex(action["data"][2:]),
         gas_price=int(action["gasPrice"], 16),
         value=int(action["value"], 16),
     )
 
+    if post_condition == {}:
```

Comment:
there should also be a check to see whether any vmException was raised or not right?

Reply:
Internally VmExceptions are caught, so we don't have an indicator on the test level. If we go for testing the instructions in the future and always go through the `Instruction.evaluate` function, we could check for all of these